Hazelcast.Net/Hazelcast.Client.Spi/HazelcastCloudDiscovery.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.RegularExpressions;
+using Hazelcast.IO;
+using Hazelcast.Logging;
+using Hazelcast.Util;
+
+namespace Hazelcast.Client.Spi
+{
+    internal class HazelcastCloudDiscovery
+    {
+        private static readonly ILogger Logger = Logging.Logger.GetLogger(typeof(HazelcastCloudDiscovery));
+
+        internal const string CloudUrlBaseProperty = "hazelcast.client.cloud.url";
+        internal const string CloudUrlBase = "https://coordinator.hazelcast.cloud";
+        private const string CloudUrlPath = "/cluster/discovery?token=";
+        private const string RegexPrivateStr = "(?<=private-address\":\").*?(?=\")";
+        private const string RegexPublicStr = "(?<=public-address\":\").*?(?=\")";
+
+        private readonly string _endpointUrl;
+        private readonly int _connectionTimeoutInMillis;
+
+        internal HazelcastCloudDiscovery(string discoveryToken, int connectionTimeoutInMillis, string cloudBaseUrl)
+        {
+            _endpointUrl = cloudBaseUrl + CloudUrlPath + discoveryToken;
+            _connectionTimeoutInMillis = connectionTimeoutInMillis;
+        }
+
+        public IDictionary<Address, Address> DiscoverNodes()
+        {
+            try
+            {
+                var uri = new Uri(_endpointUrl);
+                var httpWebRequest = (HttpWebRequest) HttpWebRequest.Create(uri);
```

Comment:
We have a special check to see if the certificate is a valid one. 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/discovery/HazelcastCloudDiscovery.java#L89
Is this needed in dotnet ?This was a request by Josef in the related prd. 
https://hazelcast.atlassian.net/wiki/spaces/EN/pages/650805252/Java+Client+Hazelcast+Cloud+Discovery+Design+Document#JavaClient:HazelcastCloudDiscoveryDesignDocument-Security

Besides the code, we will put documentation for revocation case. This is also mentioned in the same prd. I don't know if these are valid for csharp. Just reminding to not to forget.